resources/containers/oauth.button.tsx

```diff
@@ -13,20 +14,46 @@ const OAuthButton: React.FC<OAuthButtonProps> = ({
   onClick,
   label,
 }) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleClick = async (provider: GenesisOauthProvider) => {
+    setIsLoading(true);
+    try {
+      await onClick(provider);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
   return (
     <button
       data-testid={provider}
       type="button"
-      onClick={() => void onClick(provider)}
+      onClick={() => void handleClick(provider)}
       className={`${oauth_style.oauthButton} ${
-        oauth_style[`${provider}OAuthButton`]
+        !isLoading ? oauth_style[`${provider}OAuthButton`] : ""
       }`}
     >
-      <p>
-        {label} <strong>{provider}</strong>
-      </p>
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <p>
+          {label} <strong>{provider}</strong>
+        </p>
+      )}
     </button>
   );
 };
 
+const Spinner = styled.span`
```

Comment:
Im sure we have a common Spinner some where.

I dont want the client to complaint our spinners look different in each page